src/update_engine/payload_processor.cc

```diff
@@ -170,8 +173,30 @@ ActionExitCode PayloadProcessor::LoadManifest() {
 ActionExitCode PayloadProcessor::PerformOperation() {
   DCHECK(next_operation_num_ < num_total_operations_);
 
-  const InstallOperation *op = &manifest_.partition_operations(next_operation_num_);
-  DeltaPerformer *performer = &partition_performer_;
+  const size_t total_partition_operations = manifest_.partition_operations_size();
+  const InstallOperation *op = nullptr;
+  DeltaPerformer *performer = nullptr;
+
+  if (next_operation_num_ < total_partition_operations) {
+    op = &manifest_.partition_operations(next_operation_num_);
+    performer = &partition_performer_;
+  } else {
```

Comment:
Could an `else if` be used to check if it's necessary to jump into the section? The obvious check if there are any non-partition operations would also improve readability.